### PR TITLE
Avoid using instance variables in view partials

### PR DESCRIPTION
**Pull request by in-hale**
Please let me know if you think this rule should also be applied to the _top-level_ (the ones rendered from controllers) templates as well

### CODE REVIEW

**Comment by kxhitiz** [comment]
Using locals in partials (instead of instance variable) is all good.

Regarding the same for templates, I am little reluctant just because I am little concerned that how proven is it? It's something new and would love to see the gotchas and any feedbacks from the community who have been doing it or the challenges (if any) they are facing. 
Also I am not quite sure how that would look like to have the locals defined on every action, and we will have to think about other templating not just erb (maybe there are better ways). It feels like much redundancy to me.

prolly something like this:

```
def show
  my_foo = "foo" 

  render locals: { foo: my_foo } 
end
```

Saying that, if other developer agrees to go this route then I am up for it.

File: rails/no-instance-vars-in-partials.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Avoid using instance variables in view partials
+
+![](../aux_assets/images/aaron_view_instance_vars.png)
+
+## But why?
+It’s too easy to presume that certain instance variables are available in the context your partial is called in.
+
+When you first extract a partial, often for organisational reasons, it is typically used in only one place. When you reuse the partial somewhere else in your application, the controller action in which it is eventually called must correctly assign the same instance variable from the original controller action.
```
Comment (lewispb) [inline rails/no-instance-vars-in-partials.md]: Minor point but can we talk about "we" / "us" rather than "you", as this is a document for our team?

**Comment by lewispb** [comment]
Yeah I agree with what you say @kxhitiz 

I think it's good to stick with

```rb
def show
  @foo = "foo"
end
``` 

But let's use locals for partials.

**Comment by lewispb** [comment]
@in-hale thanks again for looking at this. anything else we need to do to get this merged in? :)

**Comment by in-hale** [comment]
Thanks @lewispb! I think it's good to go now

**Comment by lewispb** [comment]
Awesome thanks @in-hale

**Pull request merged**